sdk/keyvault/Azure.Security.KeyVault.Certificates/src/CertificateOperation.cs

```diff
@@ -2,17 +2,200 @@
 // Licensed under the MIT License. See License.txt in the project root for
 // license information.
 
+using System;
+using System.Globalization;
+using System.Text.Json;
+using System.Threading;
+using System.Threading.Tasks;
+
 namespace Azure.Security.KeyVault.Certificates
 {
-    public class CertificateOperation
+    public class CertificateOperation : Operation<CertificateWithPolicy>
+    {
+        private bool _hasValue = false;
+        private bool _completed = false;
+        private CertificateClient _client;
+
+        internal CertificateOperation(Response<CertificateOperationProperties> properties, CertificateClient client)
+            : base(properties.Value.Id.ToString())
+        {
+            Properties = properties;
+
+            _client = client;
+        }
+
+        public CertificateOperationProperties Properties { get; private set; }
+
+        public override bool HasCompleted => _completed;
+
+        public override bool HasValue => _hasValue;
+
+        public override Response UpdateStatus(CancellationToken cancellationToken = default)
+        {
+            if (!_completed)
+            {
+                Response<CertificateOperationProperties> pollResponse = _client.GetPendingCertificate(Properties.Name, cancellationToken);
+
+                SetRawResponse(pollResponse.GetRawResponse());
+
+                Properties = pollResponse;
+            }
+
+            if (Properties.Status == "completed")
```

Comment:
Nit: we define "magic numbers" (err, strings) elsewhere throughout code. May as well here.